qiskit/algorithms/optimizers/optimizer.py

```diff
@@ -165,7 +164,7 @@ def settings(self) -> Dict[str, Any]:
             optimizer = OptimizerClass(**settings)
 
         """
-        raise NotImplementedError
+        raise NotImplementedError("The settings method is not implemented per default.")
```

Comment:
I think it could have had a better error msg - saying something like this can be implement to allow an optimizer to be serialized for th runtime. Anyway I am not going to hold this up because of that.